types/mapcmd/map.go

```diff
@@ -57,10 +80,137 @@ func New(config *hazelcast.Config) *cobra.Command {
+		if d.Seconds() >= 1.0 {
+			return nil
+		}
+		return errors.New("ttl duration cannot be less than a second")
+	}
+	return errors.New("undefined duration type")
+}
+
+func printValueBasedOnType(cmd *cobra.Command, value interface{}) {
+	var err error
+	switch v := value.(type) {
+	case serialization.JSON:
+		if err = quick.Highlight(cmd.OutOrStdout(), fmt.Sprintln(v),
+			"json", "terminal", "tango"); err != nil {
+			cmd.Println(v.String())
+		}
+	default:
+		if v == nil {
+			cmd.Println("There is no value corresponding to the provided key")
+			break
+		}
+		cmd.Println(v)
+	}
+}
+
+func normalizeMapValue(v, vFile, vType string) (interface{}, error) {
+	var valueStr string
+	var err error
+	switch {
+	case v != "" && vFile != "":
+		return nil, hzcerrors.NewLoggableError(nil, "Only one of --value and --value-file must be specified")
+	case v != "":
+		valueStr = v
+	case vFile != "":
+		if valueStr, err = loadValueFile(vFile); err != nil {
+			err = hzcerrors.NewLoggableError(err, "Cannot load the value file. Make sure file exists and process has correct access rights")
+		}
+	default:
+		err = hzcerrors.NewLoggableError(nil, "One of the value flags (--value or --value-file) must be set")
+	}
+	if err != nil {
+		return nil, err
+	}
+	mapValue, err := internal.ConvertString(valueStr, vType)
+	if err != nil {
+		err = hzcerrors.NewLoggableError(err, "Conversion error on value %s to value-type %s", valueStr, vType)
+	}
+	return mapValue, err
+}
+
+func loadValueFile(path string) (string, error) {
+	if path == "" {
+		return "", errors.New("path cannot be empty")
+	}
+	if path == "-" {
+		if value, err := ioutil.ReadAll(os.Stdin); err != nil {
+			return "", err
+		} else {
+			return string(value), nil
+		}
+	}
+	if value, err := ioutil.ReadFile(path); err != nil {
+		return "", err
+	} else {
+		return string(value), nil
+	}
+}
+
+func isCloudIssue(err error, config *hazelcast.Config) (bool, error) {
+	isCloudCluster := config.Cluster.Cloud.Enabled
+	if networkErrMsg, handled := hzcerrors.TranslateNetworkError(err, isCloudCluster); handled {
+		err = hzcerrors.NewLoggableError(err, networkErrMsg)
+		return true, err
+	}
+	return false, err
+}
+
+func withDashPrefix(flag string, short bool) string {
+	if flag == "" {
+		return ""
+	}
+	if short {
+		return fmt.Sprintf("-%s", flag)
+	}
+	return fmt.Sprintf("--%s", flag)
+}
+
+func ObtainOrderingOfValueFlags(args []string) (vOrder []byte, tOrder []int) {
+	if len(args) == 0 {
+		return
+	}
+	for _, arg := range args {
+		vShort := withDashPrefix(MapValueFlagShort, true)
+		v := withDashPrefix(MapValueFlag, false)
+		fShort := withDashPrefix(MapValueFileFlagShort, true)
+		f := withDashPrefix(MapValueFileFlag, false)
+		tShort := withDashPrefix(MapValueTypeFlagShort, true)
+		t := withDashPrefix(MapValueTypeFlag, false)
+		switch arg {
+		case vShort, v:
+			vOrder = append(vOrder, 's')
+		case fShort, f:
+			vOrder = append(vOrder, 'f')
+		case tShort, t:
+			if len(vOrder) == 0 {
```

Comment:
I don't think the logic here is correct. `./hzc map put-all -n mm -t string -k k1 -v vk1` doesn't work.

Reply:
As I mentioned in the example section of put-all:
<img width="1097" alt="Screen Shot 2022-06-03 at 17 13 41" src="https://user-images.githubusercontent.com/43863606/171871647-8318ed58-439a-4797-9373-ab016bc7c05e.png">

Reply:
I don't think this behavior is acceptable. For now, let's assume only a single `--value-type` and a single `--key-type` may be provided by the user, which applies to all keys and values.

Reply:
The case I wrote in below explain the behavior of `put-all` and `get-all`:
- These three `put-all` denote the same thing semantically.
<img width="945" alt="Screen Shot 2022-06-04 at 15 41 44" src="https://user-images.githubusercontent.com/43863606/171999430-d1b784fb-e5db-48e8-9a39-e9d6ce506044.png">
- `get-all` output will be the same for all of them.
<img width="611" alt="Screen Shot 2022-06-04 at 15 37 52" src="https://user-images.githubusercontent.com/43863606/171999387-ee336870-1db3-4fb7-b0d7-08869862a985.png">

Reply:
For `put-all` and `get-all` behavior changed, only a single --value-type and a single --key-type may be provided by the user, which applies to all keys and values.
- These four put-all are the same semantically.
<img width="977" alt="Screen Shot 2022-06-04 at 21 44 02" src="https://user-images.githubusercontent.com/43863606/172022109-5b33e829-1c90-4e9c-82d8-2f5f28328e3b.png">